packages/phoenix-evals/src/phoenix/evals/preview/evaluators.py

```diff
@@ -189,98 +114,168 @@ def direction(self) -> DirectionType:
         """The direction for score optimization."""
         return self._direction
 
+    @property
+    def input_schema(self) -> Optional[type[BaseModel]]:
```

Comment:
I think these should be classmethods + properties so they can be read prior to instantiation